.gitignore

```diff
@@ -3,6 +3,10 @@
 ##
 ## Get latest from https://github.com/github/gitignore/blob/master/VisualStudio.gitignore
 
+
+# Config file
+appsettings.json
```

Comment:
I think that we are not setting ourselves for success here.
This will oblige the developer to rename the `appsettings.json.example` to `appsettings.json`, and it will fail if they don't and therefore they'll need to read the doc …

I would rather recommend that:
1. you don't ignore `appsettings.json` (this is the natural file that need to be in the sample)
2. but you do ignore another file that you have locally with the app coordinates you want to share: `appsettings-local.json`
3. and you change the `ReadFromJsonFile` so it reads both from both files in this order (the `appsettings-local.json` will therefore override the content of the `appsettings.json`  which is what you want.

```CSharp
 public static SampleConfiguration ReadFromJsonFile(string path)
 {
  IConfigurationRoot Configuration;

  var builder = new ConfigurationBuilder()
            .SetBasePath(Directory.GetCurrentDirectory())
            .AddJsonFile(path)
            .AddJsonFile(path.Replace(".json", ".local.json"), true);

            Configuration = builder.Build();
            return Configuration.Get<SampleConfiguration>();
        }
```
This means: no renaming, but you are still safe to develop locally.

Reply:
I'm not sure I agree here.  I understand the downside of having to rename the example file but this is a common pattern in samples across platforms.  The code you are suggesting is somewhat back magic.  I don't understand what it means to add two JsonFiles to config.  I don't know what that trailing true parameter means.  There is a risk that someone leaves that code in production code and then you have a situation where you get different behavior from a newly cloned repo that your clean normal repo.

I think I would rather the developer be aware of this explicit step to create a settings file.  Thoughts? /cc @jthake-msft

Reply:
I'm going to let @jasonjoh  drive our standard here. 
I do agree iwth your concern for sure...

Reply:
So I think I see what @jmprieur is going for here, but I'm not sure that I see any benefit. If you move the App ID into another file that is ignored, you have the same issue in that the developer still has an explicit action to create this file and add the app ID into it. If they don't, the app will still fail, but it will be a bit harder to understand why. The app will try to use "YOUR_APP_ID_HERE" as the app ID :). And of course there's always the possibility they'll just change it in appsettings.json without creating the .local file, and then inadvertently check it in, etc., etc.

It might be a good idea to add a check in the code to catch when the file is not present and explicitly return an error message referring them to the README on how to do the rename and edit.

Reply:
@jasonjoh this is the standard that has been followed in this project.

Reply:
Could we add a appSettings.json with dummy values using --force?  The reason I'm suggesting that is because when I opened a project from a clean repo I go a load of errors that all went away when I created an appSettings.json file.  The problem is that it wasn't obvious what was causing all the errors.  Initially, I thought the sample was broken.

Reply:
@jasonjoh - any thoughts on Darrel's proposal?

Reply:
@jasonjoh I suggested to Irvine that we not block moving forward on my suggestion.  But we might want to consider the pros/cons of having a example appSettings.json committed but any changes to it ignored by Git.

Reply:
I'm not sure that you can commit a file to Git and then ignore subsequent changes. If you commit a file, then add it to .gitignore after the fact, Git still tracks its changes. If there's a way to do that, great, but that's why I went with the .example approach :)

Reply:
@jasonjoh I sit corrected.  I could have sworn that's how gitignore works by default but I just tested it and I am wong.  You can tell a file to ignore all future changes using the --assume-unchanged[1] switch but that is a per-dev setting and doesn't get pushed, so that is no use to us.   Interestingly the popular answer to the problem on SO was to do exactly what Jean Marc had suggested [2].  So I'll be quiet now.

[1] https://stackoverflow.com/a/26854768/6819
[2] https://stackoverflow.com/questions/1396617/committing-machine-specific-configuration-files/1397180#1397180